lib/LANraragi/Plugin/Metadata/Chaika.pm

```diff
@@ -154,15 +157,49 @@ sub get_json_from_chaika {
 # Parses the JSON obtained from the Chaika API to get the tags.
 sub parse_chaika_json {
 
-    my ( $json ) = @_;
+    my ( $json, $addother, $addsource ) = @_;
+    
+    my $logger = get_plugin_logger();
+    $logger->debug("Chaika JSON: " . $json);
 
     my $tags = $json->{"tags"} || ();
     foreach my $tag (@$tags) {
         #Replace underscores with spaces
         $tag =~ s/_/ /g;
+        
+        #Add 'other' namespace if none
+        if ($addother && index($tag, ":") == -1) {
+            $tag = "other:" . $tag;
+        }
     }
 
-    return ( join( ', ', @$tags ), $json->{"title"} );
+    my $category = lc $json->{"category"};
+    my $download = $json->{"download"} ? $json->{"download"} : $json->{"archives"}->[0]->{"link"};
+    my $gallery = $json->{"gallery"} ? $json->{"gallery"} : $json->{"id"};
+    my $timestamp = $json->{"posted"};
+    if ( $tags ) {
```

Comment:
People might not want those additional tags, so it'd be good to gate them behind a preference.   

(Maybe this can just use `$addother` as well? I feel like if you want those extra tags you also probably want the E-H style behavior of namespacing everything)

Reply:
That's a good idea. I will add another parameter $addextra in order to make it flexible for users, even at the cost of having to carry it all the way to parse_chaika_json.😅